_posts/2021-10-01-gs_00004_basic_shell_script.md

```diff
@@ -0,0 +1,318 @@
+his/her own way. This, however, has a huge drawback. If there is a new
+colleague, he/she has to find his/her own way. Moreover, if you take over the
+project from a colleague you have to spend time to understand his/her project
+organization. A better way would be to have a project template that everyone
+uses. Let's create a very basic one!
+
+First, you usually would create a folder with the project name. Then, inside
+this folder, you might have a folder for the project sources and one for the
+documentation. Additionally, there might be a **README** and a **LICENSE** file.
+In the REAMDE you usually put basic information. For example, how you can
+install the software or how your project is organized. The LICENSE file is
+important if you want to publish your project. It tells others how they are
+allowed to use it.
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+Check our [GitHub repository](https://github.com/bitsandvolts/getting_started)
+and find an example for a README and a LICENSE file.
+
+</div>
+
+From the previous posts and the first bash script above you should already know
+how to do it. Use nano to open a file 'project_template.sh' and enter the
+following:
+
+    #!/bin/bash
+
+    # Print something to the terminal
+    echo "Greetings from Bits&Volts (bitsandvolts.org)"
+
+    # Create the project folder
+    mkdir my_project
+
+    # move into the project folder
+    cd my_project
+
+    # create other folders and files
+    mkdir sources
+    mkdir documentation
+    touch README.md
+    touch LICENSE
+
+<div class='note-normal' markdown="1">
+_Note: '.md' is a Markdown extension, which is typical for a README. We will talk about it in a future post._
+</div>
+
+Save the file, add the correct privileges and run it:
+
+    $ chmod +x project_template.sh
+    $ ./project_template.sh
+
+If you list your directory contents, you should see the created project:
+
+    $ ls
+
+So far so good! However, not all projects are called 'my_project' and it would
+be inconvenient to always change the script. That's why we want to introduce you
+to something called **arguments**. If you call your script you can give it
+additional parameters that are used by the script. In particular, we want to
+call our script like this:
+
+    $ ./project_template.sh project_name
+
+Before we can do it, we have to make a tiny change. Within the script, such
+arguments can be accessed with a
+<span>$</span>-sign and a number. <span>$</span>1 would be the first argument,
+<span>$</span>2 the second, and so on. In our script, we need to change the
+following lines:
+
+    # Create the project folder
+    mkdir $1
+
+    # move into the project folder
+    cd $1
+
+Easy, isn't it? Now try to call the script
+
+    $ ./project_template.sh project_name
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+<span>$</span>0 holds the name of the script. Add a line to the script printing
```

Comment:
Maybe explain that `$0` is in general the name/path of the program ?

Reply:
recheck